rollup/cmd/gas_oracle/app/app.go

```diff
@@ -65,35 +64,36 @@ func action(ctx *cli.Context) error {
 	registry := prometheus.DefaultRegisterer
 	metrics.Server(ctx, registry.(*prometheus.Registry))
 
-	l1client, err := ethclient.Dial(cfg.L1Config.Endpoint)
+	l1Client, err := ethclient.Dial(cfg.L1Config.Endpoint)
 	if err != nil {
 		log.Error("failed to connect l1 geth", "config file", cfgFile, "error", err)
 		return err
 	}
 
 	// Init l2geth connection
-	l2client, err := ethclient.Dial(cfg.L2Config.Endpoint)
+	l2Client, err := ethclient.Dial(cfg.L2Config.Endpoint)
 	if err != nil {
 		log.Error("failed to connect l2 geth", "config file", cfgFile, "error", err)
 		return err
 	}
 
-	l1watcher := watcher.NewL1WatcherClient(ctx.Context, l1client, cfg.L1Config.StartHeight, cfg.L1Config.Confirmations, cfg.L1Config.L1MessageQueueAddress, cfg.L1Config.ScrollChainContractAddress, db, registry)
+	l1watcher := watcher.NewL1WatcherClient(ctx.Context, l1Client, cfg.L1Config.StartHeight, cfg.L1Config.Confirmations, cfg.L1Config.L1MessageQueueAddress, cfg.L1Config.ScrollChainContractAddress, db, registry)
 
 	l1relayer, err := relayer.NewLayer1Relayer(ctx.Context, db, cfg.L1Config.RelayerConfig, registry)
 	if err != nil {
 		log.Error("failed to create new l1 relayer", "config file", cfgFile, "error", err)
 		return err
 	}
-	l2relayer, err := relayer.NewLayer2Relayer(ctx.Context, l2client, db, cfg.L2Config.RelayerConfig, false /* initGenesis */, registry)
+	l2relayer, err := relayer.NewLayer2Relayer(ctx.Context, l2Client, db, cfg.L2Config.RelayerConfig, false /* initGenesis */, registry)
 	if err != nil {
 		log.Error("failed to create new l2 relayer", "config file", cfgFile, "error", err)
 		return err
 	}
 	// Start l1 watcher process
 	go utils.LoopWithContext(subCtx, 10*time.Second, func(ctx context.Context) {
-		number, loopErr := butils.GetLatestConfirmedBlockNumber(ctx, l1client, cfg.L1Config.Confirmations)
-		if loopErr != nil {
+		// Fetch the latest block number to decrease the delay when fetching gas prices
+		number, loopErr := l1Client.BlockNumber(ctx)
+		if err != nil {
```

Comment:
why change `loopErr` to `err`?

Reply:
bug fixed in [c83d6bd](https://github.com/scroll-tech/scroll/pull/920/commits/c83d6bd5d28cb5c925e8adc27cfbfefb29654b6e)

Reply:
I don't recommend using the latest block height, how about using an independent confirm field in the config file?

Reply:
fetching L1 base bee can be a light-weight op.
btw, fetching the latest blocks seems to be better than other larger confirmation numbers, thus no need to add an extra config.